[recursor]: unable to compile 4.3.4 on Debian 10 [Makefile:1522: lwres.o] Error 1

Libboost version:

ii  libboost-atomic1.67-dev:amd64          1.67.0-13+deb10u1                   amd64        atomic data types, operations, and memory ordering constraints
ii  libboost-atomic1.67.0:amd64            1.67.0-13+deb10u1                   amd64        atomic data types, operations, and memory ordering constraints
ii  libboost-chrono1.67-dev:amd64          1.67.0-13+deb10u1                   amd64        C++ representation of time duration, time point, and clocks
ii  libboost-chrono1.67.0:amd64            1.67.0-13+deb10u1                   amd64        C++ representation of time duration, time point, and clocks
ii  libboost-context-dev:amd64             1.67.0.1                            amd64        provides a sort of cooperative multitasking on a single thread (default version)
ii  libboost-context1.67-dev:amd64         1.67.0-13+deb10u1                   amd64        provides a sort of cooperative multitasking on a single thread
ii  libboost-context1.67.0:amd64           1.67.0-13+deb10u1                   amd64        provides a sort of cooperative multitasking on a single thread
ii  libboost-date-time1.67-dev:amd64       1.67.0-13+deb10u1                   amd64        set of date-time libraries based on generic programming concepts
ii  libboost-date-time1.67.0:amd64         1.67.0-13+deb10u1                   amd64        set of date-time libraries based on generic programming concepts
ii  libboost-dev:amd64                     1.67.0.1                            amd64        Boost C++ Libraries development files (default version)
ii  libboost-program-options-dev:amd64     1.67.0.1                            amd64        program options library for C++ (default version)
ii  libboost-program-options1.67-dev:amd64 1.67.0-13+deb10u1                   amd64        program options library for C++
ii  libboost-program-options1.67.0:amd64   1.67.0-13+deb10u1                   amd64        program options library for C++
ii  libboost-serialization-dev:amd64       1.67.0.1                            amd64        serialization library for C++ (default version)
ii  libboost-serialization1.67-dev:amd64   1.67.0-13+deb10u1                   amd64        serialization library for C++
ii  libboost-serialization1.67.0:amd64     1.67.0-13+deb10u1                   amd64        serialization library for C++
ii  libboost-system-dev:amd64              1.67.0.1                            amd64        Operating system (e.g. diagnostics support) library (default version)
ii  libboost-system1.67-dev:amd64          1.67.0-13+deb10u1                   amd64        Operating system (e.g. diagnostics support) library
ii  libboost-system1.67.0:amd64            1.67.0-13+deb10u1                   amd64        Operating system (e.g. diagnostics support) library
ii  libboost-test-dev:amd64                1.67.0.1                            amd64        components for writing and executing test suites (default version)
ii  libboost-test1.67-dev:amd64            1.67.0-13+deb10u1                   amd64        components for writing and executing test suites
ii  libboost-test1.67.0:amd64              1.67.0-13+deb10u1                   amd64        components for writing and executing test suites
ii  libboost-thread-dev:amd64              1.67.0.1                            amd64        portable C++ multi-threading (default version)
ii  libboost-thread1.67-dev:amd64          1.67.0-13+deb10u1                   amd64        portable C++ multi-threading
ii  libboost-thread1.67.0:amd64            1.67.0-13+deb10u1                   amd64        portable C++ multi-threading
ii  libboost-timer1.67.0:amd64             1.67.0-13+deb10u1                   amd64        C++ wall clock and CPU process timers
ii  libboost-tools-dev                     1.67.0.1                            amd64        Boost C++ Libraries development tools (default version)
ii  libboost1.67-dev:amd64                 1.67.0-13+deb10u1                   amd64        Boost C++ Libraries development files
ii  libboost1.67-tools-dev                 1.67.0-13+deb10u1                   amd64        Boost C++ Libraries development tools


configure: Configuration summary
configure: =====================
configure: 
configure: PowerDNS Recursor configured with:  '--prefix=/usr' '--sysconfdir=/etc/powerdns' '--mandir=/usr/share/man' '--infodir=/usr/share/info' '--libdir=/usr/lib' '--libexecdir=/usr/lib' '--with-lua=lua5.3'
configure: 
configure: CC: gcc
configure: CXX: g++
configure: LD: /usr/bin/ld -m elf_x86_64
configure: CFLAGS: -fPIE -DPIE -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 --param ssp-buffer-size=4 -fstack-protector -Wall -Wextra -Wshadow -Wno-unused-parameter -g -O2 
configure: CPPFLAGS: 
configure: CXXFLAGS:  -fPIE -DPIE -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 --param ssp-buffer-size=4 -fstack-protector -Wall -Wextra -Wshadow -Wno-unused-parameter -g -O2 
configure: LDFLAGS: -Wl,-z -Wl,relro -Wl,-z -Wl,now 
configure: LIBS: -lpthread 
configure: BOOST_CPPFLAGS:  -pthread
configure: 
configure: Features enabled
configure: ----------------
configure: Lua: lua5.3
configure: OpenSSL ECDSA: yes
configure: ed25519: yes
configure: ed448: yes
configure: Protobuf: no
configure: SNMP: yes
configure: systemd: yes
configure: nod: no
configure: dnstap: yes
configure: Context library: Boost Context
configure: 
./incfiles > htmlfiles.h



lwres.cc: In function ‘int asyncresolve(const ComboAddress&, const DNSName&, int, bool, bool, int, timeval*, boost::optional<Netmask>&, boost::optional<const ResolveContext&>, const std::shared_ptr<std::vector<std::unique_ptr<RemoteLogger> > >&, const std::shared_ptr<std::vector<std::unique_ptr<FrameStreamLogger> > >&, const std::set<short unsigned int>&, LWResult*, bool*)’:
lwres.cc:287:7: error: ‘isEnabledForQueries’ was not declared in this scope
   if (isEnabledForQueries(fstrmLoggers)) {
       ^~~~~~~~~~~~~~~~~~~
lwres.cc:288:35: error: ‘queryTime’ was not declared in this scope
     logFstreamQuery(fstrmLoggers, queryTime, ip, doTCP, context ? context->d_auth : boost::none, vpacket);
                                   ^~~~~~~~~
lwres.cc:288:35: note: suggested alternative: ‘getTime’
     logFstreamQuery(fstrmLoggers, queryTime, ip, doTCP, context ? context->d_auth : boost::none, vpacket);
                                   ^~~~~~~~~
                                   getTime
lwres.cc:288:5: error: ‘logFstreamQuery’ was not declared in this scope
     logFstreamQuery(fstrmLoggers, queryTime, ip, doTCP, context ? context->d_auth : boost::none, vpacket);
     ^~~~~~~~~~~~~~~
lwres.cc:371:7: error: ‘isEnabledForResponses’ was not declared in this scope
   if (isEnabledForResponses(fstrmLoggers)) {
       ^~~~~~~~~~~~~~~~~~~~~
lwres.cc:372:95: error: ‘queryTime’ was not declared in this scope
     logFstreamResponse(fstrmLoggers, ip, doTCP, context ? context->d_auth : boost::none, buf, queryTime, *now);
                                                                                               ^~~~~~~~~
lwres.cc:372:95: note: suggested alternative: ‘getTime’
     logFstreamResponse(fstrmLoggers, ip, doTCP, context ? context->d_auth : boost::none, buf, queryTime, *now);
                                                                                               ^~~~~~~~~
                                                                                               getTime
lwres.cc:372:5: error: ‘logFstreamResponse’ was not declared in this scope
     logFstreamResponse(fstrmLoggers, ip, doTCP, context ? context->d_auth : boost::none, buf, queryTime, *now);
     ^~~~~~~~~~~~~~~~~~
make[2]: *** [Makefile:1522: lwres.o] Error 1
make[2] : on quitte le répertoire « /root/pdns-recursor-4.3.4 »
make[1]: *** [Makefile:1646: all-recursive] Error 1
make[1] : on quitte le répertoire « /root/pdns-recursor-4.3.4 »
make: *** [Makefile:1217: all] Error 2

--- a/m4/pdns_check_dnstap.m4
+++ b/m4/pdns_check_dnstap.m4
@@ -28,7 +28,7 @@ AC_DEFUN([PDNS_CHECK_DNSTAP], [
     AS_IF([test x"$FSTRM_LIBS" = "x"], [
       AC_MSG_ERROR([dnstap requested but libfstrm was not found])
     ])
-    AS_IF([test "x$PROTOBUF_LIBS" = "x" -a x"$PROTOC" = "x"], [
+    AS_IF([test "x$PROTOBUF_LIBS" = "x" -o x"$PROTOC" = "x"], [
       AC_MSG_ERROR([dnstap requested but protobuf was not found])
     ])
   ])

--- a/m4/pdns_check_dnstap.m4
+++ b/m4/pdns_check_dnstap.m4
@@ -1,4 +1,5 @@
 AC_DEFUN([PDNS_CHECK_DNSTAP], [
+  AC_REQUIRE([PDNS_WITH_PROTOBUF])
   AC_MSG_CHECKING([whether we will have dnstap])
   AC_ARG_ENABLE([dnstap],
     AS_HELP_STRING([--enable-dnstap],[enable dnstap support @<:@default=$1@:>@]),
@@ -26,6 +27,9 @@ AC_DEFUN([PDNS_CHECK_DNSTAP], [
   AS_IF([test "x$enable_dnstap" = "xyes"], [
     AS_IF([test x"$FSTRM_LIBS" = "x"], [
       AC_MSG_ERROR([dnstap requested but libfstrm was not found])
+    ])
+    AS_IF([test "x$PROTOBUF_LIBS" = "x" -a x"$PROTOC" = "x"], [
+      AC_MSG_ERROR([dnstap requested but protobuf was not found])
     ])
   ])
 ])

